corehq/util/django_migrations.py

```diff
@@ -115,42 +117,55 @@ def noop_migration():
+    version of the code. Note that this operation is never intended to succeed, because the code
+    needed for the migration no longer exists"""
+
+    def get_most_recent_migration_date():
+        return MigrationRecorder.Migration.objects \
+            .order_by('-applied') \
+            .values_list('applied') \
+            .first()[0]
 
-    return migrations.RunPython(_run_command, reverse_code=migrations.RunPython.noop, elidable=True)
+    def get_days_since_last_migration():
+        current_time = datetime.now()
+        last_migration_time = get_most_recent_migration_date()
 
+        return (current_time - last_migration_time).days
 
-def run_management_command_or_exit(command_name, *args, required_commit=None, custom_message=None, **kwargs):
-    """Helper function to run a management command automatically and abort if there was an error
-    with helpful console output to allow users to run it manually if it fails or
-    has since been removed from the codebase."""
-    try:
-        call_command(command_name, *args, **kwargs)
-    except Exception:
-        traceback.print_exc()
+    @skip_on_fresh_install
+    def _run_command(apps, schema_editor):
+        print("")
         print(f"""
-            A migration must be performed before this environment can be upgraded to the latest version
-            of CommCareHQ. This migration is run using the management command {command_name}.
-        """)
-        if required_commit or custom_message:
-            print("")
-            print(custom_message or f"""
-            Run the following commands to run the migration and get up to date:
+        This migration cannot be run, as it depends on code that has since been removed.
+        To fix this, follow the instructions below to run this migration from a previous version of our code.
+        In order to prevent this in the future, we recommend running migrations at least once every 6 weeks.
+        For reference, the current code has not run migrations for {get_days_since_last_migration()} days.
 
+        Run the following commands to run the historical migration and get up to date:
+            With a cloud setup:
                 commcare-cloud <env> fab setup_limited_release --set code_branch={required_commit}
 
-                commcare-cloud <env> django-manage --release <release created by previous command> migrate_multi
+                commcare-cloud <env> django-manage --release <release created by previous command> migrate_multi {app_name}
 
                 commcare-cloud <env> deploy commcare
-            """)
+
+            With a development setup:
+                git checkout {required_commit}
+                ./manage.py migrate {app_name}
+
+        If you are sure this migration is unnecessary, you can fake the migration:
+            With a cloud setup:
+                commcare-cloud <env> django-manage migrate --fake {app_name} {migration_name}
```

Comment:
It needs to be faked on all DBs:

```suggestion
                commcare-cloud <env> django-manage migrate_multi --fake {app_name} {migration_name}
```